src/SearchQueryService/Startup.cs

```diff
@@ -25,8 +19,11 @@ public Startup(IConfiguration configuration)
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
-
             services.AddControllers();
+
+            services.AddHttpClient();
+            var factory = services.BuildServiceProvider().GetService<IHttpClientFactory>();
+            CreateIndexes(factory);
```

Comment:
V metóde `ConfigureServices` sa majú len konfigurovať veci do DI kontajnera a nie spúšťať "výkonný" kód. Takéto metódy sa majú vykonávať v metóde `Configure`. Metóda `Configure` je pre teba výhodná aj v tom, že si ako parameter môžeš poslať čokoľvek, čo je nakonfigurované v DI kontajnery. V tvojom prípade `Indexes` (btw premenoval by som tú triedu na `IndexesProcessor`.)

Vyzeralo by to potom nejak nasledovne:

```csharp
public void ConfigureServices(IServiceCollection services)
{
  services.AddTransient<IndexesProcessor>();
  ...
}

public void Configure(IApplicationBuilder app, IWebHostEnvironment env, IndexesProcessor indexes)
{
  indexes.ProcessDirectory();
  ...
}
```

Trieda `IndexesProcessor` by dostala do seba cez konštruktor injektnutý `IHttpClient` a `IOption`, ktorý by bosahoval cestu k adresáru, ktorý má spracovať. Tento adresár by daná služba mala dostať cez ENVIRONMENT VARIABLE definované v `docker-compose` súbore. Pretože tieto veci budú danému image-u poskytované "zvnoku".

Reply:
Všetko som poprehadzoval ako bolo treba... Nateraz si ale myslím, že bude stačiť, keď v docker compose mountnem demo/indexes... Aj tak ešte možno vymyslím nejaký iný systém vyberania priečinkov.